src/Diagrams/TwoD/Path/Boolean.hs

```diff
@@ -241,6 +244,32 @@ loopExclusion :: Double -> FillRule
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
-loopExclusion tol fill path1 path2 =
+loopExclusion tol fill path1 path2 = normalizeWindings fill $
   map path2loop $ C.exclusion (map loop2path path1)
   (map loop2path path2) (fillrule fill) tol
+
+-- Force all top level loops to wind counterclockwise and revese inner loops as needed.
+normalizeWindings :: FillRule -> [Located (Trail' Loop V2 Double)] -> [Located (Trail' Loop V2 Double)]
+normalizeWindings fill = concat . map forceCC . nestedGroups fill where
+  forceCC ls | (l:_) <- ls, isClockwise l = map reverseLocLoop ls
+             | otherwise                  = ls
+
+-- Group a list of loops such that the first element of each group contains all the others.
+nestedGroups :: FillRule -> [Located (Trail' Loop V2 Double)] -> [[Located (Trail' Loop V2 Double)]]
+nestedGroups fill = map flatten . go [] where
+  go ts               []                      = ts
+  go []               (l:ls)                  = go [Node l []]                      ls
+  go ((Node n ns):ts) (l:ls) | l `contains` n = go [Node l [Node n ns]]             ls
+                             | n `contains` l = go (Node n (go ns [l]) : ts)        ls
+                             | otherwise      = go (Node n ns          : go ts [l]) ls
+  contains = containsBy fill
```

Comment:
I don't understand how this could be right with `Winding`.  If the loop contains another loop, then it must already be CC.  If it isn't CC, it will "contain" loops that are outside it.

Reply:
clock vs anticlockwise doesn't change what is contained by a loop with Winding, since it just changes the sign of the winding number and the winding rule is /= 0

Reply:
Ah, ok.  I was thinking > 0.   What is a case where you would want a different `FillRule` for this test, or in other words, why not `contains = containsBy Winding`?

Reply:
If there are self intersections in `l` or `n` it may be possible that `containsBy Winding` will be true, but `containsBy EvenOdd` will be false.